### ISSUE TITLE
Bug: NoGetOutsideModuleDefinition false positive on non-Koin get() calls

### ISSUE THREAD

**Issue by kirich1409** [androidbroadcast/Koin-Detekt#15]
## Summary

The `NoGetOutsideModuleDefinition` rule reports violations on `get()` / `getOrNull()` calls that are **not** from Koin — including Gradle build scripts, standard library calls, and other APIs that have `get()` methods.

## Current Behavior

The rule triggers on any `get()` call outside a Koin module definition, regardless of whether it's actually a Koin API call.

**Examples of false positives:**
```kotlin
// Gradle build script
val myValue = properties.get("key") // ← false positive

// Map/List access
val item = myMap.get("key") // ← false positive

// Any class with a get() method
val result = myRepository.getOrNull(id) // ← false positive
```

**Detekt output:**
```
NoGetOutsideModuleDefinition - Service locator pattern, harder to test—Use constructor injection
```

## Expected Behavior

The rule should only flag `get()` / `getOrNull()` calls that are actually from Koin's `Scope` or `ParametersHolder` APIs. Detection could be based on:
- Import statements for Koin packages (`org.koin.*`)
- Whether the call is on a Koin receiver

## Impact

**32 false positives** in a real project — the highest count of all rules.

## Note

Work on this issue has already started on branch `fix/false-positive-detection`.

## Environment

- Koin-Detekt 0.4.0
- Detekt 1.23.8
